components/org.wso2.carbon.identity.keyrotation/src/main/java/org.wso2.carbon.identity.keyrotation/dao/IdentityDAO.java

```diff
@@ -95,14 +90,14 @@ public static IdentityDAO getInstance() {
                 connection.commit();
                 while (resultSet.next()) {
                     totpSecretList
-                            .add(new TOTPSecret(resultSet.getString(TENANT_ID),
-                                    resultSet.getString(USER_NAME),
-                                    resultSet.getString(DATA_KEY),
-                                    resultSet.getString(DATA_VALUE)));
+                            .add(new TOTPSecret(resultSet.getString(KeyRotationConstants.TENANT_ID),
+                                    resultSet.getString(KeyRotationConstants.USER_NAME),
+                                    resultSet.getString(KeyRotationConstants.DATA_KEY),
+                                    resultSet.getString(KeyRotationConstants.DATA_VALUE)));
                 }
             } catch (SQLException e) {
                 connection.rollback();
-                throw new KeyRotationException("Error while retrieving TOTP secrets from IDN_IDENTITY_USER_DATA.", e);
```

Comment:
Don't we need to break the flow in this scenario?

Reply:
Yeah, we should.

Reply:
Addressed with https://github.com/wso2/identity-tools/pull/22/commits/0151a748192960c182603b555329eb3f1e0229ea